tests/remotes/s3.py

```diff
@@ -95,14 +102,72 @@ def read_text(self, encoding=None, errors=None):
 
 
 @pytest.fixture
-def s3(test_config):
+def s3_fake_creds_file(monkeypatch):
+    # https://github.com/spulec/moto#other-caveats
+    import pathlib
+
+    aws_dir = pathlib.Path("~").expanduser() / ".aws"
+    aws_dir.mkdir(exist_ok=True)
+
+    aws_creds = aws_dir / "credentials"
+    initially_exists = aws_creds.exists()
+
+    if not initially_exists:
+        aws_creds.touch()
+
+    with monkeypatch.context() as m:
+        m.setenv("AWS_ACCESS_KEY_ID", "testing")
+        m.setenv("AWS_SECRET_ACCESS_KEY", "testing")
+        m.setenv("AWS_SECURITY_TOKEN", "testing")
+        m.setenv("AWS_SESSION_TOKEN", "testing")
+        yield
+
+    if aws_creds.exists() and not initially_exists:
+        aws_creds.unlink()
+
+
+# Due to moto being uncompatible with aioboto on wrapper
+# mode, we start the moto server as a subprocess (imitates
+# a real S3 service) and then create a client that uses
+# this server.
+#
+# Originally adopted from:
+# https://github.com/dask/s3fs/blob/main/s3fs/tests/test_s3fs.py#L66-L86
+@pytest.fixture(scope="session")
+def s3_server(test_config):
     test_config.requires("s3")
-    with mock_s3():
-        import boto3
 
-        boto3.client("s3").create_bucket(Bucket=TEST_AWS_REPO_BUCKET)
+    import shlex
+    import subprocess
+
+    import requests
+
+    proc = subprocess.Popen(
+        shlex.split(f"moto_server s3 -p {TEST_AWS_S3_PORT}")
+    )
```

Comment:
Let's just use `motoserver/moto` docker image, same as we do with other remotes. (see tests/docker-compose.yml)